File: data/recomposed_projects/deepseek-coder-33b-instruct/signature/commons-graph/evosuite-test/src/test/org/apache/commons/graph/model/MutableSpanningTree_ESTest.py
from __future__ import annotations
import time
import re
import mock
import os
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.graph.Mapper import *
from src.main.org.apache.commons.graph.model.MutableSpanningTree import *

# from src.test.org.apache.commons.graph.model.MutableSpanningTree_ESTest_scaffolding import *
from src.main.org.apache.commons.graph.weight.Monoid import *
from src.main.org.apache.commons.graph.weight.primitive.IntegerWeightBaseOperations import *

# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *
# from src.main.org.evosuite.runtime.ViolatedAssumptionAnswer import *
# from src.main.org.evosuite.shaded.org.mockito.Mockito import *
# from src.main.org.evosuite.shaded.org.mockito.stubbing.Answer import *
# from src.main.org.evosuite.shaded.org.mockito.stubbing.Stubber import *


class MutableSpanningTree_ESTest(unittest.TestCase):

    def test6(self) -> None:

        integerWeightBaseOperations0 = IntegerWeightBaseOperations()
        mapper0 = mock(Mapper, ViolatedAssumptionAnswer())
        mapper0.map = Mock(return_value=None)
        mutableSpanningTree0 = MutableSpanningTree(
            integerWeightBaseOperations0, mapper0
        )
        integer0 = Integer(0)

        with pytest.raises(RuntimeError):
            mutableSpanningTree0.decorateRemoveEdge(integer0)
            verifyException(
                "org.apache.commons.graph.weight.primitive.IntegerWeightBaseOperations",
                RuntimeError,
            )

    def test5(self) -> None:

        mutableSpanningTree0 = None
        try:
            mutableSpanningTree0 = MutableSpanningTree(None, None)
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            verifyException("org.apache.commons.graph.model.MutableSpanningTree", e)

    def test4(self) -> None:

        integerWeightBaseOperations0 = IntegerWeightBaseOperations()
        mapper0 = mock(Mapper, ViolatedAssumptionAnswer())
        doReturn(None).when(mapper0).map(any_int())
        mutableSpanningTree0 = MutableSpanningTree(
            integerWeightBaseOperations0, mapper0
        )
        integer0 = Integer(0)
        mutableSpanningTree0.addVertex(integer0)
        mutableSpanningTree0.decorateAddEdge(integer0, integer0, integer0)
        integer1 = mutableSpanningTree0.getWeight()
        self.assertIsNone(integer1)

    def test3(self) -> None:

        integerWeightBaseOperations0 = IntegerWeightBaseOperations()
        integer0 = Integer(0)
        mapper0 = mock(Mapper, ViolatedAssumptionAnswer())
        mapper0.map = Mock(side_effect=lambda x: integer0)
        mutableSpanningTree0 = MutableSpanningTree(
            integerWeightBaseOperations0, mapper0
        )
        mutableSpanningTree0.decorateRemoveEdge(integer0)
        self.assertEqual(0, mutableSpanningTree0.getSize())

    def test2(self) -> None:

        integerWeightBaseOperations0 = IntegerWeightBaseOperations()
        integer0 = Integer(0)
        mapper0 = mock(Mapper, new=ViolatedAssumptionAnswer())
        mutableSpanningTree0 = MutableSpanningTree(
            integerWeightBaseOperations0, mapper0
        )
        integer1 = integerWeightBaseOperations0.append(integer0, integer0)

        with pytest.raises(RuntimeError):
            mutableSpanningTree0.decorateAddEdge(integer1, integer0, integer0)
            verifyException(
                "org.apache.commons.graph.model.BaseMutableGraph", RuntimeError
            )

    def test1(self) -> None:

        integerWeightBaseOperations0 = IntegerWeightBaseOperations()
        integer0 = Integer(0)
        integer1 = Integer(0)
        mapper0 = mock(Mapper, ViolatedAssumptionAnswer())
        when(mapper0.map(any_int())).then_return(integer0)
        mutableSpanningTree0 = MutableSpanningTree(
            integerWeightBaseOperations0, mapper0
        )
        mutableSpanningTree0.add_vertex(integer1)
        mutableSpanningTree0.decorate_add_edge(integer1, integer0, integer0)
        self.assertTrue(integer0 == integer1)

    def test0(self) -> None:

        integerWeightBaseOperations0 = IntegerWeightBaseOperations()
        integer0 = Integer(0)
        integer1 = Integer(0)
        mapper0 = mock(Mapper, ViolatedAssumptionAnswer())
        mapper0.map = Mock(side_effect=lambda x: integer0)
        mutableSpanningTree0 = MutableSpanningTree(
            integerWeightBaseOperations0, mapper0
        )
        mutableSpanningTree0.addVertex(integer0)
        mutableSpanningTree0.decorateAddEdge(integer0, integer0, integer1)
        self.assertEqual(1, mutableSpanningTree0.getOrder())

File: java_projects/cleaned_final_projects_evosuite/commons-graph/src/test/java/org/apache/commons/graph/model/MutableSpanningTree_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.apache.commons.graph.model;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.commons.graph.Mapper;
import org.apache.commons.graph.model.MutableSpanningTree;
import org.apache.commons.graph.weight.Monoid;
import org.apache.commons.graph.weight.primitive.IntegerWeightBaseOperations;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class MutableSpanningTree_ESTest extends MutableSpanningTree_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      IntegerWeightBaseOperations integerWeightBaseOperations0 = new IntegerWeightBaseOperations();
      Integer integer0 = new Integer(0);
      Integer integer1 = new Integer(0);
      Mapper<Integer, Integer> mapper0 = (Mapper<Integer, Integer>) mock(Mapper.class, new ViolatedAssumptionAnswer());
      doReturn(integer0).when(mapper0).map(anyInt());
      MutableSpanningTree<Integer, Integer, Integer> mutableSpanningTree0 = new MutableSpanningTree<Integer, Integer, Integer>(integerWeightBaseOperations0, mapper0);
      mutableSpanningTree0.addVertex(integer0);
      mutableSpanningTree0.decorateAddEdge(integer0, integer0, integer1);
      assertEquals(1, mutableSpanningTree0.getOrder());
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      IntegerWeightBaseOperations integerWeightBaseOperations0 = new IntegerWeightBaseOperations();
      Integer integer0 = new Integer(0);
      Integer integer1 = new Integer(0);
      Mapper<Integer, Integer> mapper0 = (Mapper<Integer, Integer>) mock(Mapper.class, new ViolatedAssumptionAnswer());
      doReturn(integer0).when(mapper0).map(anyInt());
      MutableSpanningTree<Integer, Integer, Integer> mutableSpanningTree0 = new MutableSpanningTree<Integer, Integer, Integer>(integerWeightBaseOperations0, mapper0);
      mutableSpanningTree0.addVertex(integer1);
      mutableSpanningTree0.decorateAddEdge(integer1, integer0, integer0);
      assertTrue(integer0.equals((Object)integer1));
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      IntegerWeightBaseOperations integerWeightBaseOperations0 = new IntegerWeightBaseOperations();
      Integer integer0 = new Integer(0);
      Mapper<Integer, Integer> mapper0 = (Mapper<Integer, Integer>) mock(Mapper.class, new ViolatedAssumptionAnswer());
      MutableSpanningTree<Integer, Integer, Integer> mutableSpanningTree0 = new MutableSpanningTree<Integer, Integer, Integer>(integerWeightBaseOperations0, mapper0);
      Integer integer1 = integerWeightBaseOperations0.append(integer0, integer0);
      // Undeclared exception!
      try { 
        mutableSpanningTree0.decorateAddEdge(integer1, integer0, integer0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.BaseMutableGraph", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      IntegerWeightBaseOperations integerWeightBaseOperations0 = new IntegerWeightBaseOperations();
      Integer integer0 = new Integer(0);
      Mapper<Integer, Integer> mapper0 = (Mapper<Integer, Integer>) mock(Mapper.class, new ViolatedAssumptionAnswer());
      doReturn(integer0).when(mapper0).map(anyInt());
      MutableSpanningTree<Integer, Integer, Integer> mutableSpanningTree0 = new MutableSpanningTree<Integer, Integer, Integer>(integerWeightBaseOperations0, mapper0);
      mutableSpanningTree0.decorateRemoveEdge(integer0);
      assertEquals(0, mutableSpanningTree0.getSize());
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      IntegerWeightBaseOperations integerWeightBaseOperations0 = new IntegerWeightBaseOperations();
      Mapper<Integer, Integer> mapper0 = (Mapper<Integer, Integer>) mock(Mapper.class, new ViolatedAssumptionAnswer());
      doReturn((Object) null).when(mapper0).map(anyInt());
      MutableSpanningTree<Integer, Integer, Integer> mutableSpanningTree0 = new MutableSpanningTree<Integer, Integer, Integer>(integerWeightBaseOperations0, mapper0);
      Integer integer0 = new Integer(0);
      mutableSpanningTree0.addVertex(integer0);
      mutableSpanningTree0.decorateAddEdge(integer0, integer0, integer0);
      Integer integer1 = mutableSpanningTree0.getWeight();
      assertNull(integer1);
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      MutableSpanningTree<Integer, Integer, Integer> mutableSpanningTree0 = null;
      try {
        mutableSpanningTree0 = new MutableSpanningTree<Integer, Integer, Integer>((Monoid<Integer>) null, (Mapper<Integer, Integer>) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.model.MutableSpanningTree", e);
      }
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      IntegerWeightBaseOperations integerWeightBaseOperations0 = new IntegerWeightBaseOperations();
      Mapper<Integer, Integer> mapper0 = (Mapper<Integer, Integer>) mock(Mapper.class, new ViolatedAssumptionAnswer());
      doReturn((Object) null).when(mapper0).map(anyInt());
      MutableSpanningTree<Integer, Integer, Integer> mutableSpanningTree0 = new MutableSpanningTree<Integer, Integer, Integer>(integerWeightBaseOperations0, mapper0);
      Integer integer0 = new Integer(0);
      // Undeclared exception!
      try { 
        mutableSpanningTree0.decorateRemoveEdge(integer0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.weight.primitive.IntegerWeightBaseOperations", e);
      }
  }
}
